File: src/two_sums_1/SolutionTest.java
package two_sums_1;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class SolutionTest {
  @Test
  void testCase1() {
    Solution solution = new Solution();
    int[] input = new int[]{ 2, 7, 11, 15 };
    int target = 9;
    int[] output = new int[]{0, 1};

    int[] result = solution.twoSum(input, target);
    assertArrayEquals(result, output);
  }
  @Test
  void testCase2() {
    Solution solution = new Solution();
    int[] input = new int[]{ 3, 2, 4 };
    int target = 6;
    int[] output = new int[]{1, 2};

    int[] result = solution.twoSum(input, target);
    assertArrayEquals(result, output);
  }
  @Test
  void testCase3() {
    Solution solution = new Solution();
    int[] input = new int[]{ 3, 3 };
    int target = 6;
    int[] output = new int[]{0, 1};

    int[] result = solution.twoSum(input, target);
    assertArrayEquals(result, output);
  }
}

File: src/two_sums_II_167/SolutionTest.kt
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.*

import two_sums_II_167.Solution

class SolutionTest {
  @Test
  fun test_case_1() {
    val input = intArrayOf(2, 7, 11, 15);
    val target = 9;
    val expected = intArrayOf(1, 2)

    val solution = Solution();
    val result = solution.twoSum(input, target);

    assertArrayEquals(expected, result);
  }
  @Test
  fun test_case_2() {
    val input = intArrayOf(2, 3, 4);
    val target = 6;
    val expected = intArrayOf(1, 3)

    val solution = Solution();
    val result = solution.twoSum(input, target);
    
    assertArrayEquals(expected, result);
  }
  @Test
  fun test_case_3() {
    val input = intArrayOf(-1, 0);
    val target = -1;
    val expected = intArrayOf(1, 2)

    val solution = Solution();
    val result = solution.twoSum(input, target);
    
    assertArrayEquals(expected, result);
  }
}